### ISSUE TITLE
[freexl] Build error

### ISSUE THREAD

**Issue by crgnam** [microsoft/vcpkg#29601]
**Host Environment**

- OS: Linux (Ubuntu 22.04.1 LTS)
- Compiler: gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0

**To Reproduce**

`./vcpkg install gdal`

**Failure logs**

```
Computing installation plan...
The following packages will be built and installed:
  * freexl[core]:x64-linux -> 1.0.6#1
    gdal[core,curl,default-features,expat,geos,gif,hdf5,iconv,jpeg,lerc,libkml,libspatialite,libxml2,lzma,netcdf,openjpeg,openssl,pcre2,png,postgresql,qhull,recommended-features,sqlite3,webp,zstd]:x64-linux -> 3.6.2#2
  * geos[core]:x64-linux -> 3.11.1
  * giflib[core]:x64-linux -> 5.2.1#3
  * hdf5[core,cpp,szip,zlib]:x64-linux -> 1.12.2#6
  * json-c[core]:x64-linux -> 2022-06-26#2
  * lerc[core]:x64-linux -> 4.0#1
  * libgeotiff[core]:x64-linux -> 1.7.1#2
  * libkml[core]:x64-linux -> 1.3.0#11
  * libpng[core]:x64-linux -> 1.6.39
  * libpq[core,lz4,openssl,zlib]:x64-linux -> 14.4#3
  * libspatialite[core,freexl,geocallbacks]:x64-linux -> 5.0.1#9
  * libwebp[core,libwebpmux,nearlossless,simd]:x64-linux -> 1.3.0
  * libxml2[core]:x64-linux -> 2.10.3
  * lz4[core]:x64-linux -> 1.9.4#1
  * netcdf-c[core,dap,hdf5,nczarr,netcdf-4,platform-default-features]:x64-linux -> 4.8.1#2
  * openjpeg[core]:x64-linux -> 2.5.0
  * pcre2[core]:x64-linux -> 10.40#1
  * pkgconf[core]:x64-linux -> 1.8.0#5
  * proj[core,net,tiff]:x64-linux -> 9.1.1
  * qhull[core]:x64-linux -> 8.0.2#4
  * sqlite3[core,rtree,tool]:x64-linux -> 3.40.1#1
  * szip[core]:x64-linux -> 2.1.1#9
  * uriparser[core]:x64-linux -> 0.9.7
  * vcpkg-cmake-get-vars[core]:x64-linux -> 2022-12-16
  * vcpkg-pkgconfig-get-modules[core]:x64-linux -> 2022-02-10#1
  * vcpkg-tool-meson[core]:x64-linux -> 0.63
  * zstd[core]:x64-linux -> 1.5.2#3
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Restored 6 package(s) from /home/crgnam/.cache/vcpkg/archives in 85.6 ms. Use --debug to see more details.
Installing 1/28 freexl:x64-linux...
Building freexl[core]:x64-linux...
-- Using cached freexl-1.0.6.tar.gz.
-- Cleaning sources at /home/crgnam/vcpkg/buildtrees/freexl/src/freexl-1-0b9ac2177e.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/crgnam/vcpkg/downloads/freexl-1.0.6.tar.gz
-- Applying patch fix-makefiles.patch
-- Applying patch fix-sources.patch
-- Applying patch fix-pc-file.patch
-- Using source at /home/crgnam/vcpkg/buildtrees/freexl/src/freexl-1-0b9ac2177e.clean
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
-- Generating configure for x64-linux
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /usr/bin/autoreconf -vfi
    Working Directory: /home/crgnam/vcpkg/buildtrees/freexl/src/freexl-1-0b9ac2177e.clean/
    Error code: 2
    See logs for more information:
      /home/crgnam/vcpkg/buildtrees/freexl/autoconf-x64-linux-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:611 (vcpkg_execute_required_process)
  ports/freexl/portfile.cmake:18 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)


error: building freexl:x64-linux failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+freexl
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[freexl]+Build+error
Include '[freexl] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: 36fb23307 2023-02-10 (22 hours ago)
```
Per the instructions: "Attach ***every failure log file*** mentioned in the console output." I included the only log file mentioned in the console output (as far as I could tell).

**Additional context**

I have tried both the `git pull` and `vcpkg update` steps (both return "Already up to date" and "No packages need updating" respectively).  This occurs both with using vcpkg directly as well as in manifest mode.

[autoconf-x64-linux-err.log](https://github.com/microsoft/vcpkg/files/10714452/autoconf-x64-linux-err.log)

**Comment by crgnam** [comment]
I just tried to run this same command (`./vcpkg install gdal`) on several systems:
- MacOS 13.2: Same error
- Windows 11 WSL (Ubuntu 20.04): INSTALLED WITHOUT ERROR

Note that in the case of the Ubuntu 22.04 (this original issue post) and the MacOS 13.2 case, an error was originally thrown stating:
```
CMake Error at scripts/cmake/vcpkg_configure_make.cmake:601 (message):
  freexl requires autoconf from the system package manager (example: "sudo
  apt-get install autoconf")
Call Stack (most recent call first):
  ports/freexl/portfile.cmake:18 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)
```

On the Ubuntu 22.04 platform I resolved that with `sudo apt-get install autoconf` while on MacOS I resolved it with `brew install autoconf`.  After running this install command for `autoconf`, the Build Error in the original issue occurred.

This seems related as it is still an issue with `freexl` (and also seems odd that it requires using the system package manager).

On the Windows 11 WSL Ubuntu 20.04 distro, no such error occurred and gdal installed without issue.

I'm wondering if perhaps I should try to fresh install of vcpkg on the problematic machines and try again.
